spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -948,6 +979,51 @@ struct ProductionVisitor
         popBuilder();
     }
 
+    void getSyncProduction(const Production& p) {
+        std::set<Production> tokens;
+
+        if ( p.isLiteral() )
+            tokens.insert(p);
+
+        else if ( auto loop = p.tryAs<production::While>() )
+            tokens = std::get<1>(loop->lookAheadProduction().lookAheads());
+
+        else if ( p.isNonTerminal() ) {
+            auto rhss = std::vector<std::vector<Production>>();
+
+            if ( const auto resolved = p.tryAs<production::Resolved>() )
+                rhss = grammar.resolved(*resolved).rhss();
+            else
+                rhss = p.rhss();
+
+            for ( const auto& rs : rhss )
+                for ( const auto& r : rs )
+                    if ( r.isLiteral() )
+                        tokens.insert(r);
+        }
+
+        if ( tokens.empty() ) {
+            hilti::logger().error("&synchronized cannot be used on field", p.location());
+            return;
+        }
+
+        auto pstate = state();
+        pstate.lahead = state().lahead;
+        pstate.lahead_end = state().lahead_end;
+        pushState(pstate);
```

Comment:
I believe this is a no-op: it just copies two fields over to itself. If anything, you probably want to clear them (though not if they could even be set here).

Reply:
I believe this is leftover code from debugging, and not adding a new state should be fine. Removing.